lib/league_stats.rb

```diff
@@ -1,40 +1,125 @@
-require_relative 'game_teams'
-require_relative 'data_loadable'
+class LeagueStats
+  attr_reader :games, :teams, :game_teams
+  def initialize(games, teams, game_teams)
+    @games = games
+    @teams = teams
+    @game_teams = game_teams
+  end
 
-class GameTeamStats
-  include DataLoadable
-  attr_reader :game_teams
+  def count_of_teams
+    @teams.count
+  end
 
-  def initialize(file_path, object)
-    @game_teams = csv_data(file_path, object)
+  def best_offense
+    team_id = unique_team_ids.max_by { |team_id| average_goals_per_team(team_id) }
+    find_name(team_id)
   end
 
-  def unique_team_ids
-    @game_teams.uniq { |game_team| game_team.team_id}.map { |game_team| game_team.team_id }
+  def worst_offense
+    team_id = unique_team_ids.min_by { |team_id| average_goals_per_team(team_id) }
+    find_name(team_id)
   end
 
-  def games_by_team(team_id)
-    @game_teams.find_all { |team| team.team_id == team_id }
+  def lowest_scoring_visitor
+    scoring('away','low')
   end
 
-  def total_games_by_team_id(team_id)
-    games_by_team(team_id).length
+  def lowest_scoring_home_team
+    scoring('home','low')
   end
 
-  def total_goals_by_team_id(team_id)
-    games_by_team(team_id).sum { |game_team| game_team.goals }
+  def highest_scoring_visitor
+    scoring('away','win')
   end
 
-  def average_goals_per_team(team_id)
-    total_goals_by_team_id(team_id).to_f / total_games_by_team_id(team_id).to_f
+  def highest_scoring_home_team
+    scoring('home','win')
   end
 
-  def best_offense
-    team_id = unique_team_ids.max_by { |team_id| average_goals_per_team(team_id) }
+  def winningest_team
+    win_ratios = Hash.new { |hash, key| hash[key] = [0,0] }
+    @games.each do |game|
+      if game.away_goals > game.home_goals
+        win_ratios[game.away_team_id][0] += 1
+      end
+      if game.home_goals > game.away_goals
+        win_ratios[game.home_team_id][0] += 1
+      end
+      win_ratios[game.away_team_id][1] += 1
+      win_ratios[game.home_team_id][1] += 1
+    end
+
+    win_percentages = win_ratios.each_with_object(Hash.new) do |(team_id, win_ratio), win_percent|
+      win_percent[team_id] = win_ratio[0].fdiv(win_ratio[1]) * 100
+    end
+
+    team_id = win_percentages.key(win_percentages.values.max)
+    find_name(team_id)
```

Comment:
This method could definitely be hung out to dry at some point.

Reply:
I'm wondering if we can use any of that logic within iteration 4/5.